From 04b95076b2b8b1230174ba8209060a2a389f1c23 Mon Sep 17 00:00:00 2001
From: Andrew Sutherland <dr3wsuth3rland@gmail.com>
Date: Sun, 1 Dec 2013 03:29:19 -0600
Subject: [PATCH 017/296] allow forcing HighEndGfx mode

people have already started fipping the lomRam prop to enable
transparent status bar, this will enable it with fewer
repercussions
---
 core/java/android/app/ActivityManager.java | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/core/java/android/app/ActivityManager.java b/core/java/android/app/ActivityManager.java
index ff68e2348b4..2a18ae39144 100644
--- a/core/java/android/app/ActivityManager.java
+++ b/core/java/android/app/ActivityManager.java
@@ -1134,7 +1134,15 @@ public class ActivityManager {
      */
     static public boolean isHighEndGfx() {
         return !isLowRamDeviceStatic() &&
-                !Resources.getSystem().getBoolean(com.android.internal.R.bool.config_avoidGfxAccel);
+                !Resources.getSystem().getBoolean(com.android.internal.R.bool.config_avoidGfxAccel)
+                || isForcedHighEndGfx();
+    }
+
+    /**
+     * @hide
+     */
+    public static boolean isForcedHighEndGfx() {
+        return SystemProperties.getBoolean("persist.sys.force_highendgfx", false);
     }
 
     /**
-- 
2.11.0

